common/templates/general.go

```diff
@@ -240,6 +241,29 @@ func CreateMessageSend(values ...interface{}) (*discordgo.MessageSend, error) {
 				ContentType: "text/plain",
 				Reader:      &buf,
 			}
+
+		case "image":
+			if val == nil {
+				return nil, errors.New("need an argument for the image")
+			}
+			str := fmt.Sprint(val)
+			img, err := http.Get(str)
+			if err != nil {
+				return nil, errors.New("error while trying to get the image")
+			}
+
+			imageName := "image.png"
+			imageType := "image/png"
+			if strings.HasSuffix(strings.ToLower(str), ".gif") {
+				imageName = "image.gif"
+				imageType = "image/gif"
+			}
+			msg.File = &discordgo.File{
+				Name: imageName,
+				ContentType: imageType,
+				Reader: img.Body,
+			}
+
```

Comment:
I think you should img.Body.Close() here.

Reply:
Why though? It works without it! :)

Reply:
Just because "it works", it doesn't mean it is right :)

As per Go's http package: `The client *must* close the response body when finished with it`

Reply:
indeed. I'll do it then! Thanks Pedro.